### PR TITLE
[FIX] Reveal SRP or Private Key wrong password error

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_

_1. What is the reason for the change?_
While trying to reveal the SRP or Private Key, if the user enters the wrong password the app would still display the "Hold To Reveal" component.

_2. What is the improvement/solution?_
This PR validates the password before allowing the user to proceed.

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

https://user-images.githubusercontent.com/17601467/182950360-9889dc69-f36c-4be2-bb2e-35c11df77a78.mov

**Issue**

Progresses https://github.com/MetaMask/metamask-mobile/issues/4781

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -111,7 +111,7 @@
     "@keystonehq/bc-ur-registry-eth": "^0.7.7",
     "@keystonehq/metamask-airgapped-keyring": "^0.3.0",
     "@keystonehq/ur-decoder": "^0.3.0",
-    "@metamask/controllers": "^30.0.2",
+    "@metamask/controllers": "git+https://github.com/MetaMask/controllers.git#37e8490212c0a0e37823fd0d8bd70175dfd38006",
```
Comment (gantunesr) [inline package.json:114]: Update controllers before merge

**Review by Cal-L — COMMENTED** [review]
Was there another issue where you can still hit return on the keyboard, which will prompt the hold to reveal modal even with an incorrect password?
Reference to where this is mentioned - https://github.com/MetaMask/metamask-mobile/issues/4781#:~:text=The%20%22Return%22%20button%20on%20the%20keyboard%20acts%20the%20same%20as%20the%20%22Next%22%20button%2C%20so%20leads%20to%20the%20%22hold%20to%20reveal%22%20dialog

**Comment by gantunesr** [comment]
In reference to https://github.com/MetaMask/metamask-mobile/pull/4799#pullrequestreview-1062591772

You're right @Cal-L, I just added an improvement in [440cfa7](https://github.com/MetaMask/metamask-mobile/pull/4799/commits/440cfa76519627a66e8120687340720a17639847)

**Review by Cal-L — APPROVED** [review]
LGTM

**Review by cortisiko — APPROVED** [review]
This is 🌮 🌮 🌮 

To test follow the accepted behavior listed [here](https://github.com/MetaMask/metamask-mobile/issues/4781#issue-1324966832)

**Pull request merged**